mmrotate/datasets/dota.py

```diff
@@ -1,361 +1,143 @@
+                    (147, 116, 116), (0, 0, 255)]
+    }
 
     def __init__(self,
-                 ann_file,
-                 pipeline,
-                 version='oc',
-                 difficulty=100,
-                 **kwargs):
-        self.version = version
-        self.difficulty = difficulty
-
-        super(DOTADataset, self).__init__(ann_file, pipeline, **kwargs)
-
-    def __len__(self):
-        """Total number of samples of data."""
-        return len(self.data_infos)
-
-    def load_annotations(self, ann_folder):
-        """
-            Args:
-                ann_folder: folder that contains DOTA v1 annotations txt files
-        """
+                 img_shape: Tuple[int, int] = (1024, 1024),
+                 diff_thr: int = 100,
+                 **kwargs) -> None:
+        self.img_shape = img_shape
+        self.diff_thr = diff_thr
+        super().__init__(**kwargs)
+
+    def load_data_list(self) -> List[dict]:
+        """Load annotations from an annotation file named as ``self.ann_file``
+        Returns:
+            List[dict]: A list of annotation.
+        """  # noqa: E501
         cls_map = {c: i
-                   for i, c in enumerate(self.CLASSES)
+                   for i, c in enumerate(self.metainfo['CLASSES'])
                    }  # in mmdet v2.0 label is 0-based
-        ann_files = glob.glob(ann_folder + '/*.txt')
-        data_infos = []
-        if not ann_files:  # test phase
-            ann_files = glob.glob(ann_folder + '/*.png')
-            for ann_file in ann_files:
+        data_list = []
+        if self.ann_file == '':
+            img_files = glob.glob(
+                osp.join(self.data_prefix['img_path'], '*.png'))
+            for img_path in img_files:
                 data_info = {}
-                img_id = osp.split(ann_file)[1][:-4]
-                img_name = img_id + '.png'
-                data_info['filename'] = img_name
-                data_info['ann'] = {}
-                data_info['ann']['bboxes'] = []
-                data_info['ann']['labels'] = []
-                data_infos.append(data_info)
+                data_info['img_path'] = img_path
+                img_name = osp.split(img_path)[1]
+                data_info['file_name'] = img_name
+                img_id = img_name[:-4]
+                data_info['img_id'] = img_id
+                data_info['height'] = self.img_shape[0]
+                data_info['width'] = self.img_shape[1]
+
+                instances = []
+                instance = {}
+                instance['bbox'] = []
+                instance['bbox_label'] = []
+                instance['ignore_flag'] = 0
+                instances.append(instance)
+                data_info['instances'] = instances
+                data_list.append(data_info)
+
+            return data_list
         else:
-            for ann_file in ann_files:
+            txt_files = glob.glob(osp.join(self.ann_file, '*.txt'))
```

Comment:
ann_file is a folder, it is different from the BaseDataset. Need to add its docstring in __init__ .